pennylane/resource/specs.py

```diff
@@ -208,6 +208,11 @@ def specs_qnode(*args, **kwargs) -> list[dict] | dict:
             info = specs_from_tape(tape, compute_depth)
             info["num_device_wires"] = len(qnode.device.wires or tape.wires)
             info["num_tape_wires"] = tape.num_wires
+            info["max_work_wires"] = (
+                None
+                if qnode.device.wires is None
+                else len(set(qnode.device.wires) - set(tape.wires))
+            )
```

Comment:
Could we do this as a separate feature instead? I could see this needing some thought and debate, and would rather not block having the testing in.